frontend/src/lib/hooks/useElementSize.ts

```diff
@@ -4,46 +4,52 @@ import type { Size2D } from "@lib/utils/geometry";
 import { elementIsVisible } from "@lib/utils/htmlElementUtils";
 
 export function useElementSize(ref: React.RefObject<HTMLElement>): Size2D {
-    const [size, setSize] = React.useState<Size2D>({ width: 0, height: 0 });
-
-    React.useEffect(() => {
-        let isHidden = false;
-        let currentSize = { width: 0, height: 0 };
-        const handleResize = (): void => {
-            let newSize: Size2D = { width: 0, height: 0 };
-            if (ref.current) {
-                // If element is not visible do not change size as it might be expensive to render
-                if (!elementIsVisible(ref.current)) {
-                    isHidden = true;
-                    return;
+    const [size, setSize] = React.useState<Size2D>({
+        width: ref.current?.offsetWidth ?? 0,
+        height: ref.current?.offsetHeight ?? 0,
+    });
+
+    React.useEffect(
+        function layoutEffect() {
```

Comment:
Should you still call the named function `layoutEffect`. I see from history that you added `React.useLayoutEffect` and then  changed back to regular `useEffect`